include/deal.II/numerics/matrix_tools.h

```diff
@@ -232,23 +234,23 @@ namespace MatrixCreator
    *
    * See the general documentation of this namespace for more information.
    */
-  template <int dim, typename number, int spacedim>
+  template <int dim, int spacedim, typename number>
   void create_mass_matrix (const Mapping<dim, spacedim>       &mapping,
                            const DoFHandler<dim,spacedim>    &dof,
                            const Quadrature<dim>    &q,
                            SparseMatrix<number>     &matrix,
-                           const Function<spacedim> *const a = 0,
+                           const Function<spacedim,number> *const a = 0,
```

Comment:
that's correct that one could generally use, say, `SparseMatrix<float>` with `Function<dim,double>`. However i think we already made this choice when we reworked `FEValues` last spring https://github.com/dealii/dealii/pull/918 where, for example, the second `number` template argument of `get_function_values` was removed in favour of `typename InputVector::value_type`. 

I see two options here: either to ask it to be consistent (which is the same to what we already did) or to introduce extra template parameter (which is IMHO an overkill).

I guess in 99% it is going to be used with `double` everywhere.